src/System.Net.WebProxy/tests/WebProxyTest.cs

```diff
@@ -181,11 +182,21 @@ public static void WebProxy_BypassOnLocal_MatchesExpected(Uri destination, bool
         {
             Uri proxyUri = new Uri("http://microsoft.com");
 
-            Assert.Equal(isLocal, new WebProxy(proxyUri, true).IsBypassed(destination));
-            Assert.False(new WebProxy(proxyUri, false).IsBypassed(destination));
+            try
+            {
+                Assert.Equal(isLocal, new WebProxy(proxyUri, true).IsBypassed(destination));
+                Assert.False(new WebProxy(proxyUri, false).IsBypassed(destination));
 
-            Assert.Equal(isLocal ? destination : proxyUri, new WebProxy(proxyUri, true).GetProxy(destination));
-            Assert.Equal(proxyUri, new WebProxy(proxyUri, false).GetProxy(destination));
+                Assert.Equal(isLocal ? destination : proxyUri, new WebProxy(proxyUri, true).GetProxy(destination));
+                Assert.Equal(proxyUri, new WebProxy(proxyUri, false).GetProxy(destination));
+            }
+            catch (SocketException exception)
+            {
+                // On Unix, getaddrinfo returns host not found, if all the machine discovery settings on the local network
+                // is turned off. Hence dns lookup for it's own hostname fails.
```

Comment:
Do we have any way to validate that we only get this exception in this circumstance?  Otherwise it seems like a GetProxy implementation that always threw SocketException.HostNotFound on Unix would pass our tests.

Reply:
GetProxy or IsBypassed wouldn't return HostNotFound in any other situation, looking at the implementation.

Reply:
What if we introduce a bug in our code?
Is there a way to detect the situation and check it is indeed the troublesome machine setting?

Reply:
Uhh, i guess i can try to run Dns.GetHostEntry(Dns.GetHostName()) and confirm that it does return host not found in the catch block. Basically run the same product code in the tester.